File: backend-python/app/models/album.py
from sqlalchemy import Column, Integer, String, Date
from app.database import Base

class Album(Base):
    __tablename__ = "album"
    
    id = Column(Integer, primary_key=True, index=True)
    pet_id = Column(Integer, nullable=False, index=True)
    image_path = Column(String, nullable=False)
    year = Column(Integer, nullable=False)
    upload_date = Column(Date)
    caption = Column(String)
    location = Column(String)
    media_type = Column(String)
    category = Column(String)

File: backend/src/main/java/com/petcare/model/Album.java
package com.petcare.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "album")
@Data
public class Album {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "pet_id", nullable = false)
    private Long petId;
    
    @Column(name = "image_path", nullable = false)
    private String imagePath;
    
    @Column(nullable = false)
    private Integer year;
    
    @Column(name = "upload_date")
    private LocalDate uploadDate;
    
    private String caption;
    
    private String location;
    
    @Column(name = "media_type")
    private String mediaType; // photo, video
    
    private String category;
}
